src/Magento/FunctionalTestingFramework/Util/TestGenerator.php

```diff
@@ -1483,7 +1483,7 @@ private function resolveStepKeyReferences($input, $actionGroupOrigin, $matchAll
                 . $stepKey . $testInvocationKey . "', 'field', 'test')";
 
             if (strpos($output, $stepKeyVarRef) !== false) {
-                $output = str_replace($stepKeyVarRef, $stepKeyVarRef . $testInvocationKey, $output);
+                $output = preg_replace('/\B\\' .$stepKeyVarRef. '\b/', $stepKeyVarRef . $testInvocationKey, $output);
```

Comment:
Insert a comment with an explanation about what's going on in this preg_replace like
```
// only replace where full word matches exactly
// ex. testVar => $testVar  but not $testVar2
```

Reply:
added